File: web-portal-bff/src/api/preferences/model/Preferences.ts
export interface RetryPolicy {
  maximumAttempts: number,
  backoffCoefficient: number
  initialInterval: String
  maximumInterval: String
  startToCloseTimeout: String
}

export interface GatewayTimeout {
  gatewayTimeoutSeconds: number,

}

export interface ServerPreferences {
  gatewayTimeoutSeconds: GatewayTimeout;
  retryPolicy: RetryPolicy;
}

export interface TimeUnit {
  value: number;
  unit: string;
}

export interface ClientPreferences {
  maximumAttempts: number;
  backoffCoefficient: number;
  gatewayTimeout: TimeUnit,
  initialInterval: TimeUnit
  maximumInterval: TimeUnit
  startToCloseTimeout: TimeUnit
}

File: integration-tests/src/test/java/io/github/cameronward301/communication_scheduler/integration_tests/model/Preferences.java
package io.github.cameronward301.communication_scheduler.integration_tests.model;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class Preferences {
    private Integer gatewayTimeoutSeconds;
    private RetryPolicy retryPolicy;

    @Data
    @Builder
    public static class RetryPolicy {
        private Integer maximumAttempts;
        private Float backoffCoefficient;
        private String initialInterval;
        private String maximumInterval;
        private String startToCloseTimeout;
    }
}
